src/directives/routeClass.js

```diff
@@ -0,0 +1,28 @@
+class routeClassDirective {
+  constructor(Route) {
+
+    this.restrict = 'A';
+
+    this.link = (scope, iElement, iAttrs) => {
+      scope.$watch(() => {
+        const routeClassDefinition = scope.$eval(iAttrs['routeClass'])
```

Comment:
what does this actually look like when used ? Given the attr appears to be a json object a comment on usage would be helpful

    <foo route-class="{ viewName: 'main', bindingName: 'binding2', className: 'bold-me' }> 

Also on that thread of thought, what about a simplified interface

    <foo route-class="if main:binding2 then boldme" }> 

thats kinda weird to see in HTML but something to think about ...